src/App.tsx

```diff
@@ -1,24 +1,205 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Notification } from './components/Notification';
+import { Header } from './components/Header';
 import { UserWarning } from './UserWarning';
+import {
+  addTodos, completeTodo, deleteTodo, getTodos, renameTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorNotice } from './types/ErrorNotice';
+import { Filter } from './types/Filter';
+import { getVisibleTodos } from './helper/getVisibleTodos';
 
-const USER_ID = 0;
+const USER_ID = 6762;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setMessageError] = useState('');
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [title, setNewTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inProcessing, setProcessingIDs] = useState<number[]>([0]);
+  const hasError = !!errorMessage || errorMessage !== '';
+  const isDisableInput = !tempTodo;
+
+  const showError = (message: string) => {
+    setMessageError(message);
+    setTimeout(() => setMessageError(''), 3000);
+  };
+
+  const loadingTodos = async () => {
+    try {
+      const todos = await getTodos(USER_ID);
+
+      setTodos(todos);
+    } catch (error) {
+      showError(ErrorNotice.LOADING);
+      setTodos(state => [...state]);
```

Comment:
Why do you need to reassign the same state? it is possible to remove this action? seems to be redundant